### PR TITLE
[Refine] group all ota metadata related logic to ota_metadata.py, implement abstraction layer

### PR DESCRIPTION
**Pull request by Bodong-Yang**
## Introduction
This PR groups all logic related to downloading, processing and parsing OTA metadata and metafiles to `ota_metadata.py`, implements an abstraction layer between detailed ota metafiles processing logic and otaclient internal.
Also `ota_metadata.py` is well documented in source code now.

## Motivation
Previously, the downloading/processing/parsing OTA metadata/metafiles are separated across different modules, like downloading `metadata.jwt` and metafiles logic are in `ota_client.py`, `create_standby` module loads original downloaded text based metafiles by its own.
With this PR, now all the detailed logic related to OTA metadata/metafiles are implemented within `ota_metadata` module, and `ota_metadata` module provides a stable internal API for other otaclient modules to use.

## Major changes
1. ota_metadata: now handles downloading/processing/parsing OTA metadata/metafiles while provides a stable API for otaclient internal use.
2. ota_metadata: internal refactoring, re-arranging and documenting.
3. ota_metafiles.proto: internal used metafiles(regular, directory, symlink, persistent) are described by protobuf, corresponding wrappers are defined for internal used.
4. otaclient now will utilize `/run/otaclient` to store runtime data.

## Minor changes
1. [Bug fix] fix downloader not properly handles empty file.
2. create_standby: not use `UpdateMeta` anymore, instead directly passing parameters.
3. downloader: supports configuring proxy and cookies via methods.
4. otaclient: only configure proxy and cookies once when update start, not passing proxy and cookies on every downloading.
5. config: now `DOWNLOADER_CONNPOOL_SIZE_PER_THREAD` is set to `20` to fully utilize the `MAX_CONCURRENT_DOWNLOAD_TASKS`.

## TODO
- [x] implement test files
- [x] pass local test 
- [x] vm test

### CODE REVIEW

**Review by obi-t4 — COMMENTED** [review]
I left some comments, please have a check.

**Comment by obi-t4** [comment]
@Bodong-Yang 
The complexity of the ota client seems to be increasing with each PR.
I think there is something fundamentally wrong with this.

File: otaclient/app/create_standby/common.py [hunk 1]
```diff
@@ -131,57 +130,25 @@ def get_tracker(
...
-class RegularDelta(Dict[str, RegularInfSet]):
-    """Dict[str, RegularInfSet]"""
+class RegularDelta(Dict[bytes, Set[RegularInf]]):
+    """Dict[bytes, Set[RegularInf]]"""
 
     def __init__(self):
-        super().__init__()
-        # for fast lookup regularinf entry
-        self._pathset: Set[Path] = set()
+        self._path_set: Set[str] = set()
```
[the remark this answers is not in the document]
Comment (Bodong-Yang) [reply otaclient/app/create_standby/common.py:137]: Sorry about that, I will take care of it in the next time.

File: otaclient/app/downloader.py [hunk 2]
```diff
@@ -249,14 +257,18 @@ def _download_task(
         _start_time = time.thread_time_ns()
 
         # special treatment for empty file
-        if dst == self.EMPTY_STR_SHA256 and not (dst_p := Path(dst)).is_file():
-            dst_p.write_bytes(b"")
+        if digest == self.EMPTY_STR_SHA256:
```
[the remark this answers is not in the document]
Comment (Bodong-Yang) [reply otaclient/app/downloader.py:260]: Yes... and this bug was introduced a lot time ago, but it doesn't impact too much as we now only download once for each unique file, so this bug only results in download empty file one more time for current otaclient.

[the remark this answers is not in the document]
Comment (Bodong-Yang) [reply otaclient/app/downloader.py:260]: Yes... and this bug was introduced a lot time ago, but it doesn't impact too much as we now only download once for each unique file, so this bug only results in download empty file one more time for current otaclient.

File: otaclient/app/proto/streamer.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+"""
+
+
+from ._common import MessageType, MessageWrapperType
+from typing import Iterable, Optional, Type, Generic, BinaryIO
+
+UINT32_LEN = 4  # bytes
+
+
+class Uint32LenDelimitedReader:
```
[the remark this answers is not in the document]
Comment (Bodong-Yang) [reply otaclient/app/proto/streamer.py:30]: Storing all metafiles in memory will take a lot of memory(~200M). It is very common that program stores runtime data in the file system, also considering that we have devices that don't have so much memory(raspberry pi).

**Review by obi-t4 — APPROVED** [review]
LGTM :+1:

**Review by obi-t4 — APPROVED** [review]

**Pull request merged**